Functions/Admin/Set-RsEmailConfiguration.ps1

```diff
@@ -0,0 +1,175 @@
+        $EmailCredential,
+        
+        [Alias('SqlServerInstance')]
+        [string]
+        $ReportServerInstance,
+        
+        [Alias('SqlServerVersion')]
+        [ReportingServicesTools.SqlServerVersion]
+        $ReportServerVersion,
+        
+        [string]
+        $ComputerName,
+        
+        [System.Management.Automation.PSCredential]
+        $Credential
+    )
+    
+    #region Connect to Report Server using WMI
+    try
+    {
+        $splat = @{ }
+        if ($PSBoundParameters.ContainsKey('ReportServerInstance')) { $splat['ReportServerInstance'] = $ReportServerInstance }
+        if ($PSBoundParameters.ContainsKey('ReportServerVersion')) { $splat['ReportServerVersion'] = $ReportServerVersion }
+        if ($PSBoundParameters.ContainsKey('ComputerName')) { $splat['ComputerName'] = $ComputerName }
+        if ($PSBoundParameters.ContainsKey('Credential')) { $splat['Credential'] = $Credential }
+        $rsWmiObject = New-RsConfigurationSettingObject @splat
+    }
+    catch
+    {
+        throw
+    }
+    #endregion Connect to Report Server using WMI
+    
+    #region Backwards compatibility
+    if (-not $PSBoundParameters.ContainsKey("Authentication"))
+    {
+        $CallName = (Get-PSCallStack)[0].InvocationInfo.InvocationName
+        
+        if ($CallName -eq "Set-RsEmailSettingsAsNoAuth") { $Authentication = "None" }
+        if ($CallName -eq "Set-RsEmailSettingsAsBasicAuth") { $Authentication = "Basic" }
+        if ($CallName -eq "Set-RsEmailSettingsAsNTLMAuth") { $Authentication = "Ntlm" }
+    }
+    #endregion Backwards compatibility
+    
+    if (($Authentication -like "Basic") -and (-not $EmailCredential))
+    {
+        throw (New-Object System.Management.Automation.PSArgumentException("Basic authentication requires passing credentials using the 'EmailCredential' parameter!"))
+    }
+    
+    #region Handle Credentials on basic authentication
+    $UserName = ''
+    $Password = ''
+    
+    if ($Authentication -like "Basic")
+    {
+        $UserName = $EmailCredential.UserName
+        $Password = $EmailCredential.GetNetworkCredential().Password
+    }
+    #endregion Handle Credentials on basic authentication
+    
+    try { $result = $rsWmiObject.SetAuthenticatedEmailConfiguration($true, $SmtpServer, $SenderAddress, $UserName, $Password, $Authentication.Value__, $true) }
+    catch { throw (New-Object System.Exception("Failed to update email settings: $($_.Exception.Message)", $_.Exception)) }
+    
+    if ($result.HRESULT -ne 0)
+    {
+        throw "Failed to update email settings. Errocode: $($result.HRESULT)"
+    }
+}
+
+# Backwards compatiblity
+New-Alias -Name "Set-RsEmailSettingsAsNTLMAuth" -Value Set-RsEmailConfiguration -Scope Global
+New-Alias -Name "Set-RsEmailSettingsAsNoAuth" -Value Set-RsEmailConfiguration -Scope Global
+New-Alias -Name "Set-RsEmailSettingsAsBasicAuth" -Value Set-RsEmailConfiguration -Scope Global
```

Comment:
This does not achieve backwards compatibility. Right now if someone tries running Set-RsEmailSettingsAsNoAuth, the would expect RS Email Settings to be configured with No Authentication but when this command actually runs, $Authentication will be NTLM.